projects/ion/src/lib/notification/service/notification.service.ts

```diff
@@ -0,0 +1,202 @@
+    private injector: Injector
+  ) {}
+
+  public success(
+    title: string,
+    message: string,
+    options?: NotificationConfigOptions
+  ): void {
+    if (!this.notificationContainerComponentRef)
+      this.createNotificationContainer();
+
+    const notification = this.createNotificationInstance();
+
+    this.configNotification(notification.instance, title, message, options);
+
+    notification.hostView.detectChanges();
+    notification.changeDetectorRef.detectChanges();
+
+    this.notificationContainerComponentRef.instance.addNotification(
+      notification
+    );
+
+    this.notificationContainerComponentRef.changeDetectorRef.detectChanges();
+
+    return this.componentSubscriber.next();
+  }
+
+  public info(
+    title: string,
+    message: string,
+    options?: NotificationConfigOptions
+  ): void {
+    if (!this.notificationContainerComponentRef)
+      this.createNotificationContainer();
+
+    const notification = this.createNotificationInstance();
+
+    this.configNotification(
+      notification.instance,
+      title,
+      message,
+      options,
+      NOTIFICATION_TYPES.info
+    );
+
+    notification.hostView.detectChanges();
+    notification.changeDetectorRef.detectChanges();
+
+    this.notificationContainerComponentRef.instance.addNotification(
+      notification
+    );
+
+    this.notificationContainerComponentRef.changeDetectorRef.detectChanges();
+
+    return this.componentSubscriber.next();
+  }
+
+  public warning(
+    title: string,
+    message: string,
+    options?: NotificationConfigOptions
+  ): void {
+    if (!this.notificationContainerComponentRef)
+      this.createNotificationContainer();
+
+    const notification = this.createNotificationInstance();
+
+    this.configNotification(
+      notification.instance,
+      title,
+      message,
+      options,
+      NOTIFICATION_TYPES.warning
+    );
+
+    notification.hostView.detectChanges();
+    notification.changeDetectorRef.detectChanges();
+
+    this.notificationContainerComponentRef.instance.addNotification(
+      notification
+    );
+
+    this.notificationContainerComponentRef.changeDetectorRef.detectChanges();
+
+    return this.componentSubscriber.next();
+  }
+
+  public error(
+    title: string,
+    message: string,
+    options?: NotificationConfigOptions
+  ): void {
+    if (!this.notificationContainerComponentRef)
+      this.createNotificationContainer();
+
+    const notification = this.createNotificationInstance();
+
+    this.configNotification(
+      notification.instance,
+      title,
+      message,
+      options,
+      NOTIFICATION_TYPES.negative
+    );
+
+    notification.hostView.detectChanges();
+    notification.changeDetectorRef.detectChanges();
+
+    this.notificationContainerComponentRef.instance.addNotification(
+      notification
+    );
+
+    this.notificationContainerComponentRef.changeDetectorRef.detectChanges();
+
+    return this.componentSubscriber.next();
+  }
```

Comment:
the success, information, warning and error functions are practically the same. I believe you can reduce code repetition by creating, for example, a generic function.

Reply:
I waiting this to approve this awesome PR :)